### ISSUE TITLE
Exposing to the web

### ISSUE THREAD

**Issue by kevinahuber** [mshenfield/web-pi-cam#5]

**Comment by kevinahuber** [comment]
@mshenfield This may be a little trickier than I thought. Because it is not just exposing the raspberry pi, it would be exposing my entire network, right?

**Comment by mshenfield** [comment]
@kahjav yes - but you can safely only forward traffic to port 80 on the pi if you want - the world won't know about any other computer behind the router. I think it is still kind of a dangerous proposition - if someone has access to the Pi they have access to the rest of your network if the compromise the Pi. Also we wouldn't be able to bring it to like an event and set it up unless we had access to their router :wink:

I'm basically repeating what I learned from this article about using your Rasberry Pi as a home web server. http://raspberrywebserver.com/serveradmin/get-your-raspberry-pi-web-site-on-line.html

If we wanted to dynamically route stuff to the server, wherever we set it up, we might be able to script something with [localtunnel](https://localtunnel.me/) and a beacon to broadcast the url. It would be sub-domained to `*.localhost.me`, and maybe have to change each time we ran it. I still want to understand more about how `localtunnel` works before I'm totally for it, but if neither of us have time to learn it, f**k it, I'd still give it a try.

**Comment by kevinahuber** [comment]
Ah that's tricky. We want something that we could just plug and play. I'll try to get a holed punched sometime soon so you can access it.

**Comment by kevinahuber** [comment]
What if we hosted the app elsewhere (AWS or the sort), then had the camera connect into the app as the camera source? Then the raspberry pi would just need a strong internet connection. May create a greater lag in the camera, but would greatly increase mobility.
